### PR TITLE
Add send_batch_topic function for batch topic routing

### PR DESCRIPTION
**Pull request by gdwoolbert3**
## Summary

Adds `send_batch_topic()` function to enable batch message sending with topic-based routing, similar to how `send_batch()` extends `send()` for standard queues.

This PR is raised to address part of #455. The bulk of the work was completed by @smyrgeorge, this merely adds a batch version of the new `send_topic` function.

### New Function: `send_batch_topic()`

Combines the routing logic of `send_topic()` with the batch sending capability of `send_batch()`.

**Signatures:**
- `send_batch_topic(routing_key, msgs[])`
- `send_batch_topic(routing_key, msgs[], headers[])`
- `send_batch_topic(routing_key, msgs[], delay)`
- `send_batch_topic(routing_key, msgs[], headers[], delay)`

**Returns:** `TABLE(queue_name text, msg_id bigint)` - Tracks which messages were sent to which queues

### Implementation Details

- **Main function** (4 parameters): Uses topic pattern matching to route batch messages to all matching queues
- **Function overloads**: 3 convenience wrappers for different parameter combinations
- **Validation**: Routing key, message array (NULL/empty), and delay (negative) checks
- **Performance**: Uses pre-compiled regex patterns with indexed lookups
- **Transaction safety**: All sends succeed or all fail (atomic)

### Files Modified

- `pgmq-extension/sql/pgmq.sql` - Main implementation (4 functions)
- `pgmq-extension/sql/pgmq--1.10.0--1.11.0.sql` - Migration file (identical functions)
- `pgmq-extension/test/sql/topic_tests.sql` - Comprehensive test suite (9 test cases)
- `pgmq-extension/test/expected/topic_tests.out` - Expected test output
- `docs/api/sql/functions.md` - Complete API reference
- `docs/topics.md` - User guide with usage patterns
- `pgmq-extension/README.md` - Updated features list

## Testing

All tests pass (4/4 test suites):
- ✅ base tests
- ✅ fifo_rr_tests  
- ✅ fifo_tests
- ✅ topic_tests (including 9 new send_batch_topic tests)

**Test coverage:**
- Multi-queue routing (fan-out to 2+ queues)
- Headers functionality
- Delay/visibility timing
- Empty results (no pattern matches)
- Single queue edge case
- NULL/empty array validation
- Negative delay validation
- Invalid routing key validation

## Documentation

Comprehensive documentation added:

### API Reference (functions.md)
- Complete "Topic-Based Routing" section
- All 4 `send_batch_topic()` signatures
- Parameter tables and return type descriptions
- Multiple working examples
- Also documented `send_topic()` and helper functions

### User Guide (topics.md)
- Detailed function documentation with usage notes
- New usage pattern: "Batch Message Publishing"
- Examples showing all parameter combinations
- Behavior notes (transactional, fan-out, edge cases)

### README
- Added "Topic-based routing" to features list

## Example Usage

```sql
-- Send batch of notifications to matching queues
SELECT * FROM pgmq.send_batch_topic(
    'notify.email.welcome',
    ARRAY[
        '{"to": "user1@example.com", "name": "Alice"}',
        '{"to": "user2@example.com", "name": "Bob"}'
    ]::jsonb[]
);

-- Returns (if 2 queues match):
--  queue_name         | msg_id
-- --------------------+--------
--  email_queue        | 1
--  email_queue        | 2
--  all_notifications  | 1
--  all_notifications  | 2
```

## Compatibility

- Follows existing PGMQ patterns (matches `send_batch` and `send_topic` conventions)
- No breaking changes
- Backward compatible with all existing functionality

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pgmq-extension/sql/pgmq.sql [hunk 1]
```diff
@@ -1852,3 +1852,86 @@ BEGIN
...
+$$
+DECLARE
+    b RECORD;
+BEGIN
+    PERFORM pgmq.validate_routing_key(routing_key);
+
+    IF msgs IS NULL OR array_length(msgs, 1) IS NULL THEN
+        RAISE EXCEPTION 'msgs cannot be NULL or empty';
+    END IF;
+
```
[the remark this answers is not in the document]
Comment (gdwoolbert3) [reply pgmq-extension/sql/pgmq.sql]: f22345c05ca2f08fb73373a5973d3c37188dabee

**Comment by smyrgeorge** [comment]
@gdwoolbert3 @ChuckHend 
Yes the `select distinct` seems right, my bad

**Review by ChuckHend — APPROVED** [review]
Thank you @gdwoolbert3.
